python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifier.java

```diff
@@ -0,0 +1,125 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.quickfix;
+
+import com.sonar.sslr.api.AstNode;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import org.sonar.plugins.python.api.IssueLocation;
+import org.sonar.plugins.python.api.PythonCheck;
+import org.sonar.plugins.python.api.PythonCheck.PreciseIssue;
+import org.sonar.plugins.python.api.PythonFile;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.PythonVisitorContext;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.python.SubscriptionVisitor;
+import org.sonar.python.parser.PythonParser;
+import org.sonar.python.quickfix.IssueWithQuickFix;
+import org.sonar.python.semantic.ProjectLevelSymbolTable;
+import org.sonar.python.tree.PythonTreeMaker;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PythonQuickFixVerifier {
+  private PythonQuickFixVerifier() {
+  }
+
+  public static void verifyNoQuickFix(PythonCheck check, String code){
+    List<PythonCheck.PreciseIssue> issuesWithQuickFix = PythonQuickFixVerifier.getIssuesWithQuickFix(check, code);
+    assertThat(issuesWithQuickFix).isEmpty();
+  }
+
+  public static void verify(PythonCheck check, String codeWithIssue, String codeFixed) {
+    List<PythonCheck.PreciseIssue> issues = PythonQuickFixVerifier
+      .getIssuesWithQuickFix(check, codeWithIssue);
+
+    assertThat(issues).hasSize(1);
+    IssueWithQuickFix issue = (IssueWithQuickFix) issues.get(0);
+
+    assertThat(issue.getQuickFixes()).hasSize(1);
+
+    String codeQFApplied = PythonQuickFixVerifier.applyQuickFix(codeWithIssue, issue);
+    assertThat(codeQFApplied).isEqualTo(codeFixed);
+  }
+
+  private static List<PreciseIssue> scanFileForIssues(PythonCheck check, PythonVisitorContext context) {
+    check.scanFile(context);
+    if (check instanceof PythonSubscriptionCheck) {
+      SubscriptionVisitor.analyze(Collections.singletonList((PythonSubscriptionCheck) check), context);
+    }
+    return context.getIssues();
+  }
+
+  public static List<PreciseIssue> getIssuesWithQuickFix(PythonCheck check,String codeWithIssue) {
+    PythonParser parser = PythonParser.create();
+    PythonQuickFixFile pythonFile = new PythonQuickFixFile(codeWithIssue);
+    AstNode astNode = parser.parse(pythonFile.content());
+    FileInput parse = new PythonTreeMaker().fileInput(astNode);
+
+    PythonVisitorContext visitorContext = new PythonVisitorContext(parse,
+      pythonFile, null, "",
+      ProjectLevelSymbolTable.empty());
+
+    return scanFileForIssues(check, visitorContext);
+  }
+
+  public static String applyQuickFix(String codeWithIssue, IssueWithQuickFix issueWithQuickFix) {
+    IssueLocation loc = issueWithQuickFix.getQuickFixes().get(0).getTextEdits().get(0).issueLocation;
```

Comment:
I would prefer to have the assertion about the number of quick fixes just above this line so that we can be sure that `get(0)` cannot fail.